CustomerService.UnitTests/Service/CustomerServiceTests.cs

```diff
@@ -0,0 +1,174 @@
+﻿using CustomerService.DBContext;
+using CustomerService.Models;
+using CustomerService.Repository;
+using CustomerService.Service;
+using IO.Eventuate.Tram;
+using IO.Eventuate.Tram.Events.Publisher;
+using IO.Eventuate.Tram.Events.Subscriber;
+using IO.Eventuate.Tram.Local.Kafka.Consumer;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using ServiceCommon.Classes;
+using System;
+
+namespace CustomerService.UnitTests.Service
```

Comment:
A unit test should be purely in-memory - not database or message broker.

The unit tests for CustomerService should test the CustomerService class with mocks for CustomerRepository and DomainEventPublisher.

There should be a component test that uses the database and message broker:

* Run Docker container
* POST /customers to create a customer
* Publishes OrderCreated event to trigger credit reservation
* Asserts that a CreditReserved (or Failed) event for that order was published in response.

Reply:
Unit tests for CustomerService with mocks for repository and DomainEventPublisher are pending